### DISCUSSION TITLE
Cant use nested fiel resolvers with typegoose

### DISCUSSION THREAD

**Discussion by angelhodar** [MichalLytek/type-graphql#1447] [Q&A]
Hi, I have just run into a situation where the integration middleware with typegoose has a problem with field resolvers in subdocuments. I have this example model:

```ts
@ObjectType()
@pre<User>('find', function () {
  this.populate('interests')
})
@pre<User>('findOne', function () {
  this.populate('interests')
})
export class User {
  @Field(() => [Interest])
  @prop({ ref: () => Interest})
  interests: Ref<Interest>[];

  @Field(() => Boolean)
  hasImportantInterest(): boolean {
    if (!isDocumentArray(this.interests)) return false
    return this.interests.some((interest) => interest.isImportant());
  }
```

Now the problem is that this line:

```ts
if (!isDocumentArray(this.interests)) return false
```
always returns false because the typegoose middleware has already converted the interests documents into plain objects so the ``ìsImportant`` function is not present. Here I paste the typegoose middleware for reference:

```ts
import { Model, Document } from "mongoose";
import { getClass } from "@typegoose/typegoose";
import { MiddlewareFn } from "type-graphql";

export const TypegooseMiddleware: MiddlewareFn = async (_, next) => {
  const result = await next();

  if (Array.isArray(result)) {
    return result.map(item => (item instanceof Model ? convertDocument(item) : item));
  }

  if (result instanceof Model) {
    return convertDocument(result);
  }

  return result;
};

function convertDocument(doc: Document) {
  const convertedDocument = doc.toObject();
  const DocumentClass = getClass(doc)!;
  Object.setPrototypeOf(convertedDocument, DocumentClass.prototype);
  return convertedDocument;
}
```

This could be solved by applying the middleware **AFTER**  all the field resolvers have been executed, but I have no idea how to do that or if its even possible to check the resolver stack position. Thanks in advance!!

**Comment by MichalLytek** [comment]
I would suggest looking for a way to transform graphql result before sending the result as json response in docs of graphql server you're using.
TypeGraphQL can't hook into execution of the graphql schema, it's just a schema builder.

**Reply by angelhodar** [reply]
Hi Michal, I am currently using apollo server v4. So if I do that the typegoose middleware wouldnt be necessary? If I understand correctly you are suggesting to apply the logic of the typegoose middleware directly into the graphql server to only execute that logic at the end of the resolver chain?

**Reply by MichalLytek** [reply]
TBH I don't use typegoose at all.
If I remember correctly, the middleware is designed to solve the problem that TypeGraphQL tries to create instance of the `@Root()` object class, which conflicts with the mongoose Document instance.
So if you disable the middleware, some weird issues might occur.
